tests/integration/components/item-component-test.js

```diff
@@ -0,0 +1,54 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { click, find, render } from '@ember/test-helpers';
+import { run } from '@ember/runloop';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+
+module('Integration | Component | item-component', function(hooks) {
+  setupRenderingTest(hooks);
+
+  const TEMPLATE = hbs`{{item-component
+    item=item
+    markAsRead=(action markAsRead item)
+  }}`;
+
+  let createItem = function(owner, data) {
+    let store = owner.lookup('service:store');
+    return run(function() {
+      return store.createRecord('item', data);
+    });
+  };
+
+  let setUpAndRender = function(testContext, properties) {
+    testContext.set('item', properties.item);
+    testContext.set('markAsRead', properties.markAsRead || function() {});
+
+    return render(TEMPLATE);
+  };
+
+  test('it renders the content of the item', async function(assert) {
+    const item = createItem(this.owner, {
+      title: 'Hello world'
+    });
+
+    await setUpAndRender(this, { item });
+
+    assert.equal(find('[data-test-item-title]').innerText, item.get('title'));
+  });
+
+  test('it triggers markAsRead action with item when hide button is clicked', async function(assert){
+    const item = createItem(this.owner);
+    const markAsRead = sinon.spy();
+
+    await setUpAndRender(this, {
+      item,
+      markAsRead
+    });
+
+    await click('[data-test-mark-as-read]');
+
+    assert.ok( markAsRead.calledWith( item ) );
```

Comment:
There's an interesting thing happening at this point - we're testing that the `markAsRead` action is being called with `item` but this doesn't actually happen in the component, it happens in the invoker, and in this case on L13 of this file... This means that we're testing the test in a way. 

To overcome this we could add a test to item-list-component where we click on a particular item and ensure that markAsRead is called with that item.

Reply:
Makes sense, I'll move this test up.